taca_ngi_pipeline/deliver/deliver_castor.py

```diff
@@ -0,0 +1,170 @@
+    """
+    def __init__(self, projectid=None, sampleid=None, **kwargs):
+        super(CastorProjectDeliverer, self).__init__(
+            projectid,
+            sampleid,
+            **kwargs)
+        # castor specific fields
+        self.castordeliverypath = getattr(self, 'castordeliverypath', None)
+        self.castorsftpserver = getattr(self, 'castorsftpserver', None)
+        self.castorsftpserver_user = getattr(self, 'castorsftpserver_user', None)
+    
+
+    def deliver_project(self):
+        """ Deliver all samples in a project to castor
+            
+            :returns: True if all samples were delivered successfully, False if
+                any sample was not properly delivered or ready to be delivered
+        """
+        try:
+            logger.info("Delivering {} to {}".format(
+                str(self), self.expand_path(self.castordeliverypath)))
+            if self.get_delivery_status() == 'DELIVERED' \
+                    and not self.force:
+                logger.info("{} has already been delivered".format(str(self)))
+                return True
+            # right now, don't catch any errors since we're assuming any thrown
+            # errors needs to be handled by manual intervention
+            status = True
+            # Open sftp session with castor, in this way multiple tranfer will be possible
+            try:
+                transport=paramiko.Transport(self.castorsftpserver)
+                password = getpass.getpass(prompt='Bianca/Castor Password for user {}:'.format(self.castorsftpserver_user))
+                transport.connect(username = "{}-{}".format(self.castorsftpserver_user, self.uppnexid), password = password)
+            except Exception as e:
+                logger.error("Caught exception: {}: {}".format(e.__class__, e))
+                raise
+
+            #memorize all samples that needs to be delivered
+            samples_to_deliver = [sentry['sampleid'] for sentry in db.project_sample_entries(
+                    db.dbcon(), self.projectid).get('samples', [])]
+            status = True
+            #open the sftp client
+            #sftp_client = transport.open_sftp_client()
+            sftp_client = MySFTPClient.from_transport(transport)
+            # move to the delivery directory in the sftp
+            #open one client session and leave it open for all the time of the transfer
+            sftp_client.chdir(self.expand_path(self.castordeliverypath))
+            #create the project folder in the remote server
+            if not rexists(sftp_client, self.projectid):
+                sftp_client.mkdir(self.projectid)
+            #move inside the project folder
+            sftp_client.chdir(self.projectid)
+            #now cycle across the samples
+            for sampleid in samples_to_deliver:
+                sampleDelivererObj = CastorSampleDeliverer(self.projectid, sampleid, sftp_client)
```

Comment:
@b97pla the only method that I rewrote is `do_delivery` that does not set statuses, everything else is done with the class in `SampleDeliverer` . So what happens is identical to what happens now